### PR TITLE
add bin support for JDK 6 and before

### PR DESCRIPTION
**Pull request by coreen**
using linux download as no option for macos in those legacy versions

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ sudo: false
 env:
   - EXPECTED_VERSION=1.8.0_74 JDKW_VERSION=8u74 JDKW_BUILD=b02 JDKW_VERBOSE=1
   - EXPECTED_VERSION=1.8.0_121 JDKW_VERSION=8u121 JDKW_BUILD=b13 JDKW_VERBOSE=1 JDKW_TOKEN=e9e7ea248e2c4826b92b3f075a80e441
+  - EXPECTED_VERSION=1.6.0_38 JDKW_VERSION=6u38 JDKW_BUILD=b05
```
Comment (vjkoskela) [inline .travis.yml]: Please add `JDKW_VERBOSE=1` to this in case it needs to be debugged

File: jdk-wrapper.sh [hunk 2]
```diff
@@ -189,6 +189,10 @@ extension="tar.gz"
 if [ "${JDKW_PLATFORM}" = "macosx-x64" ]; then
   extension="dmg"
 fi
+# JDK 6 and before use .bin extension, note no specific macosx download available for these versions (using linux)
```
Comment (vjkoskela) [inline jdk-wrapper.sh]: So you're saying versions equal to and prior to JDK6 for Linux only used a bin format? And only a bin format?

It looks like it was a bin but that also wrapped an rpm:

http://www.oracle.com/technetwork/java/javase/downloads/java-archive-downloads-javase6-419409.html

Can you make the comment a little clearer?

File: jdk-wrapper.sh [hunk 3]
```diff
@@ -189,6 +189,10 @@ extension="tar.gz"
...
+if [ "${JDKW_VERSION:2:1}" -le "6" ]; then
+  extension="bin"
```
Comment (vjkoskela) [inline jdk-wrapper.sh]: If my understanding above is correct then you should nest this an os check for linux before doing the version check and then default to bin if both are true.

Please also add an asterisk to line 71 and describe this exception as a foot-note in the header comment and do the same in the Configuration section of the README.md file.

File: jdk-wrapper.sh [hunk 4]
```diff
@@ -189,6 +189,10 @@ extension="tar.gz"
 if [ "${JDKW_PLATFORM}" = "macosx-x64" ]; then
   extension="dmg"
 fi
+# JDK 6 and before use .bin extension, note no specific macosx download available for these versions (using linux)
+if [ "${JDKW_VERSION:2:1}" -le "6" ]; then
```
Comment (vjkoskela) [inline jdk-wrapper.sh]: This doesn't work.

I think you're looking for the prefix pattern match to digits. For example, if the Java version were ever 60u110 you wouldn't want to treat it as version 6. So we need to either extract the `60` (numeric prefix) or dump the suffix (starting at `u`).

Something like `${JDKW_VERSION%%u*}` should do it but I haven't test it thoroughly.

**Review by vjkoskela — CHANGES_REQUESTED** [review]

**Comment by vjkoskela** [comment]
Thanks this is great!

File: jdk-wrapper.sh (hunk 2 above)
Comment (coreen) [reply jdk-wrapper.sh]: There are 2 options for download, a regular `.bin` and an rpm-wrapped `.bin`. I've updated the comment to mention the optionally wrapped rpm as another supported option for the `.bin` extension download.

File: jdk-wrapper.sh (hunk 3 above)
Comment (coreen) [reply jdk-wrapper.sh]: Adding the `.bin` extensions here was so even MacOS users would be able to use JDK 6 and before versions. Nesting under Linux will nullify that.

**Comment by coreen** [comment]
Turns out Oracle pushed a change recently that forces users to login in order to download JDK 6 and 7 versions, so this PR will no longer work. Announcement in screenshot below or at http://www.oracle.com/technetwork/java/javase/archive-139210.html
screen shot 2017-05-31 at 6 32 10 pm

**Pull request closed without merging**